### DISCUSSION TITLE
Organization Metrics in Base Activity for User

### DISCUSSION THREAD

**Discussion by w33ts** [lowlighter/metrics#1396] [Q&A]
Hello!

I apologize if this was right in front of me and I missed it but I was wondering if I'm doing something wrong here. I'd like to include private organization metrics in my "Activity" section here: 
![image](https://user-images.githubusercontent.com/27108677/223480298-2e3c298b-c38a-4e0f-8a90-f3843edcf6fc.png)

However, it doesn't seem to be showing. It does _appear_ to have my commits but not my PRs or reviews. I ensured I had the following settings in my yml, just to be sure:
![image](https://user-images.githubusercontent.com/27108677/223480673-50e51a7b-eaa9-4138-847a-54e7c93d4d06.png)

And my token _does_ have the read:org scope:
![image](https://user-images.githubusercontent.com/27108677/223480981-0a2b4aaf-c8ca-4756-bba3-8a0c84946f78.png)

Am I doing something wrong here or is including those metrics not possible?

Thanks again!

**Comment by lowlighter** [comment]
Hi 👋 !

It may or may not be supported but I'm not sure to be honest.
The query used doesn't have any filter, but it's possible that GitHub may filter out organizations stats by default.

Could you try to run the following queries in the [GitHub GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer):

### Stats from 1st january (should have 0 PR reviews since it's roughly what is used currently by this app)
```graphql
{
  viewer {
    contributionsCollection(from: "2023-01-01T05:00:00.000Z") {
      totalPullRequestContributions
      totalPullRequestReviewContributions
      totalIssueContributions
      totalCommitContributions
    }
  }
}
```

### Stats from 1st january with organization filter

#### Fetch organization id
```graphql
{
  organization(login: "organization-login") {
    id
  }
}
```

#### Stats from 1st january with organization filter
```graphql
{
  viewer {
    contributionsCollection(from: "2023-01-01T05:00:00.000Z", organizationID: "organization-id-from-previous-request") {
      totalPullRequestContributions
      totalPullRequestReviewContributions
      totalIssueContributions
      totalCommitContributions
    }
  }
}
```

____

If the later returns non-zero values for several of your organization and it seems correct, maybe it's just missing a loop on organization to fetch data on them, so we could patch the base plugin to track these missing data

**Reply by w33ts** [reply]
Hey lowlighter, thanks for the reply!

So I ran the first query and this is the return:
![image](https://user-images.githubusercontent.com/27108677/225685910-6d68f097-89fa-4fc4-963e-bacdaabd8be7.png)

Here is the second query for the specific organization with the most PRs, etc:
![image](https://user-images.githubusercontent.com/27108677/225686142-59984a4b-ade4-446f-ab32-ca0d4ecc487b.png)


I fully expect the issues section to be zero, as we use JIRA for our issue tracking.

I suppose it needs that loop you mentioned to include organization data.

**Reply by lowlighter** [reply]
Nice ! 

Does the sum of both results looks plausible to you ?
Because they look similiar according to your screenshots, and the totalPullRequestReview is even the same, so I'm wondering if the first might actually already contains the second result 🤔

**Reply by w33ts** [reply]
Yeah, that looks about right and I suspect it does. That organization holds the vast majority of my activity, truthfully. The rest is probably the commits/PRs to add the metrics to my profile and maybe a couple of other things.
